elide-core/src/main/java/com/yahoo/elide/core/PersistentResource.java

```diff
@@ -1737,10 +1700,7 @@ protected void addInverseRelation(String relationName, Object inverseObj) {
                 // retrieve the object id to return to the caller. If no fields on the object are readable by the caller
                 // then they will be filtered out and only the id is returned. Similarly, all future requests to this
                 // object will behave as expected.
-                boolean isMutation = resource.getRequestScope().isMutatingMultipleEntities();
-                if (!isMutation || !resource.getRequestScope().getNewResources().contains(resource)) {
-                    resource.checkFieldAwarePermissions(permission);
-                }
+                resource.checkReadPermissionsAllowNewEntities();
```

Comment:
`filter()` use to call `checkFieldAwarePermissions()` which check permission on field level. Is it enough that `checkReadPermissionsAllowNewEntities()` only check permission entity on entity level?

Reply:
The method checkFieldAwarePermissions(permission) in turn calls only the checkPermission() method which adds only entity level checks.